### ISSUE TITLE
ExceptionsManager.js:180 Unhandled promise rejection RangeError: Maximum call stack size exceeded when consume api

### ISSUE THREAD

**Issue by keyneslim** [infinitered/ignite#1705]
What's going on?

Having ExceptionsManager.js:180 Unhandled promise rejection RangeError: Maximum call stack size exceeded

Steps to reproduce

1. Init new Pizza App
2. Go into demo list
3. Press refresh to trigger the demo list one or two times
4. ExceptionsManager.js:180 Unhandled promise rejection RangeError: Maximum call stack size exceeded is in the log
<img width="791" alt="Screenshot 2021-06-18 at 10 36 30 PM" src="https://user-images.githubusercontent.com/28332346/122577980-026fab80-d086-11eb-92cf-8d3cf85aa02b.png">

`ignite doctor` results:
System
  platform           darwin                                                
  arch               x64                                                   
  cpu                12 cores     Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz 
  directory          PizzaApp     /Users/xx/Projects/PizzaApp         

JavaScript (and globally-installed packages)
  node               16.0.0       /usr/local/bin/node 
  npm                7.10.0       /usr/local/bin/npm  
    @angular/cli     11.2.11                          
    http-server      0.12.3                           
    npm              7.10.0                           
    yarn             1.22.10                          
  yarn               1.22.10      /usr/local/bin/yarn 

Ignite
  ignite-cli         6.12.2       /Users/xx/.npm/_npx/e31027f3785124a8/node_modules/.bin/ignite      
  ignite src         build        /Users/xx/.npm/_npx/e31027f3785124a8/node_modules/ignite-cli/build 

Android
  java               1.8.0_292    /usr/bin/java 
  android home       -            undefined     

iOS
  xcode              12.5       
  cocoapods          1.10.1       /usr/local/bin/pod

**Comment by osiloke** [comment]
I have the same issue but with the latest ignite as shown below. It's a very strange issue.

When a call is made using apisauce, it executes but does not return to the mobx state tree.
However, if you console log response, it returns to the mobx state tree but gives the `maximum call stack error`

```
System
  platform           darwin
  arch               x64
  cpu                4 cores      Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
  directory          docheck      /Users/***/workspace/***/***

JavaScript (and globally-installed packages)
  node                 14.17.0      /Users/***/.nvm/versions/node/v14.17.0/bin/node
  npm                  6.14.13      /Users/***/.nvm/versions/node/v14.17.0/bin/npm
    apollo             2.33.4
    appcenter-cli      2.9.0
    expo-cli           4.5.2
    ignite-cli         7.0.3
    json2csv           5.0.6
    live-server        1.2.1
    nhost              0.4.7
    npm                6.14.13
    passwordgen        0.3.0
    react-native-cli   2.0.1
    sqlite3            4.2.0
    yarn               1.22.10
  yarn                 1.22.10      /Users/***/.nvm/versions/node/v14.17.0/bin/yarn
    create-react-app   3.4.1
    ignite-cli         2.2.4

Ignite
  ignite-cli         7.0.3        /Users/***/.nvm/versions/node/v14.17.0/bin/ignite
  ignite src         build        /Users/***/.nvm/versions/node/v14.17.0/lib/node_modules/ignite-cli/build

Android
  java               12.0.1       /usr/bin/java
  android home       -            /Users/***/Library/Android/sdk/

iOS
  xcode              12.5
  cocoapods          1.10.1       /usr/local/bin/pod
```

**Comment by osiloke** [comment]
The problem seems to stem from the API layer and apisauce. If you make calls using fetch, this error does not come up.

**Comment by keyneslim** [comment]
I'm also not sure about this. I also facing issue of MST Store is not working as expected in Release Mode, compare to debug mode. Guess i will let this boiler plate go :(

**Comment by osiloke** [comment]
This is a great boilerplate, however, there seems to be an issue with async and apisauce (axios) and use fetch in the api. If you make the method a sync method it should work.

i.e
```
getUsers(){
    return fetch(this.api.config.url, {
      method: "post",
      body: JSON.stringify(getEvents()),
    })
      .then((resp) => resp.json())
      .then((data) => ({
        kind: "ok",
        data,
      }))
  }
}
```

**Comment by imxeno** [comment]
Any updates on this? I'm probably facing the same issue.

**Comment by Arnoldnuo** [comment]
Any updates on this? I'm facing the same issue too.

**Comment by osiloke** [comment]
> Any updates on this? I'm facing the same issue too.

You can use my workaround to bypass this bug as shown in the snippet below. The only difference here is that getUsers() is not an async function but returns an async function.

> This is a great boilerplate, however, there seems to be an issue with async and apisauce (axios) and use fetch in the api. If you make the method a sync method it should work.
> 
> i.e
> 
> ```
> getUsers(){
>     return fetch(this.api.config.url, {
>       method: "post",
>       body: JSON.stringify(getEvents()),
>     })
>       .then((resp) => resp.json())
>       .then((data) => ({
>         kind: "ok",
>         data,
>       }))
>   }
> }
> ```

**Comment by Apophenian** [comment]
I'm also having this issue and unfortunately osiloke's workaround is not working for me.

**Comment by imxeno** [comment]
**If you are not using expo,** adding `@babel/plugin-transform-async-to-generator` to `babel.config.js` is a working workaround. If you *are* using expo, you're unfortunately out of luck.

**Comment by Apophenian** [comment]
This worked for me.  Thank you.

**Comment by TrevPennington** [comment]
Facing this issue (using Expo)

**Comment by sitanjob** [comment]
> Using Expo here and adding @babel/plugin-transform-async-to-generator to babel.config.js worked.

The issue still happen

**Comment by iamdavidmartin** [comment]
> Using Expo here and adding @babel/plugin-transform-async-to-generator to babel.config.js worked.

Also using Expo, this didn't fix is. My console is full of these errors.

It is indeed very strange. Like @osiloke, I have to console.log to get the API calls to complete. However, I'm not using apisauce, I'm using `amplify-js` which uses axios internally.

I've tried tracking this error down but I have no idea where to find the issue and the error logs are not helpful at all.
![image](https://user-images.githubusercontent.com/373633/132787082-2c127cdf-e9b7-45a7-8d67-cb4e5a718eaa.png)

**Comment by chreck** [comment]
I have this issue also with the latest ignite boilerplate version 7.5.0. Just apply the patch file on the new created app.
[0001-test-useEffect-and-api-call.patch.zip](https://github.com/infinitered/ignite/files/7201703/0001-test-useEffect-and-api-call.patch.zip)

**Steps to reproduce**
1. Init new app
2. Apply patch
3. Run in debug mode

**Post comment**
The API call is made and the data is stored in the in the mobx-state-tree. And this is only occurring in the debug mode. As long as the data is always stored in the state-tree I think this error log can be ignored.

**Comment by poplaw** [comment]
I've managed to fix this (just as a workaround) by removing the <ToggleStorybook /> form app.tsx. For some reason, it seems that the Storybook wrapper is causing the issue.

**Comment by thanhlmm** [comment]
@poplaw It id the trick

**Comment by WalrusSoup** [comment]
> I've managed to fix this (just as a workaround) by removing the `<ToggleStorybook />` from app.tsx. For some reason, it seems that the Storybook wrapper is causing the issue.

I'm going to piggyback this. As a newcomer to react native with this boilerplate, I ended up wasting 4 hours of my life because my generator that called an API would not return to the caller. I don't know why this fixed it.

**Comment by memyselfandm** [comment]
Also worked for me. in case anyone else was having issues, this had been giving me a fit when trying to get `expo-in-app-purchases` to work as well

**Comment by niieani** [comment]
I'm having the same issue with `@nhost/hasura-auth-js` crashing. 
Removing `<ToggleStorybook />` as a workaround also did the trick.

**Comment by dannyhw** [comment]
@jamonholmgren I think maybe this issue could be closed since it looks like its fixed by #1906

**Comment by nesjett** [comment]
Just some info if someone needs storybook and still facing issues: If I remove < ToggleStorybook > as proposed, it works, but thats just a workaround, the real issue is caused because I'm executing my code "too early" in a useEffect() just at the app startup and that has some conflict with storybook initialization (I guess).

If I delay that useEffect() action (in my case I'm launching a mobx action to call the server API inside of It)  for example with a TimeOut of 500ms, everything works well.

**Comment by yulolimum** [comment]
Our team has decided to move away from Storybook altogether. We will be removing it in [Ignite v8: Maverick](https://github.com/infinitered/ignite/milestone/19). Meanwhile, the reported issue should be resolved with #1906 (thank you!); and for users that prefer not to patch, removing the `<ToggleStorybook>` wrapper and mounting Storybook inside a navigation screen should achieve similar results.
